plugin/include/runtime-support.h

```diff
@@ -557,15 +557,19 @@ inline ::zeek::ValPtr to_val(const T& t, ::zeek::TypePtr target, const std::stri
 }
 
 /**
- * Converts a Spicy-side tuple to a Zeek record value. The result is returned
+ * Converts a Spicy-side enum to a Zeek record value. The result is returned
  * with ref count +1.
  */
 template<typename T, typename std::enable_if_t<std::is_enum<T>::value>*>
 inline ::zeek::ValPtr to_val(const T& t, ::zeek::TypePtr target, const std::string& location) {
     if ( target->Tag() != ::zeek::TYPE_ENUM )
         throw TypeMismatch("enum", target, location);
 
-    return zeek::compat::EnumTypeGetEnumVal(target->AsEnumType(), static_cast<int>(t));
+    // Zeek's enum can't be negative, so we swap in max_int for our Undef (-1).
+    auto it = static_cast<int64_t>(t);
+    bro_int_t bt = (it >= 0 ? it : std::numeric_limits<::bro_int_t>::max());
```

Comment:
Hmm, it could be possible that a user specified an enum value with max_int for which we would now emit at least semantically incorrect code. It would be great if we could add some validation for that to declare that particular value "magic" and off-limits to users (ideally here, but I am not sure that kind of reflection is possible; else in Spicy proper).

Reply:
I agree in principle, though in practice bro_int_t is a 64-bit integer and I'd be surprised if anybody ever defined that max as an enum value. Question is if we want to put this knowledge about bro_int_t into Spicy proper for a validation there.

Reply:
As a middle ground we could add a runtime check so that we do not silently mutilate data, e.g.,

```.cpp 
assert(((void)"Serializing enum values with value max_int to Zeek is not supported",
          it == std::numeric_limits<int64_t>::max()));
```

Reply:
assert is debug mode only, I'll put a runtime exception in

Reply:
I lost track of what changed here since my initial comment, but the current code first does a possibly lossy conversion of `t: uint64_t` to `it: int64_t`, and in the creation of `bt: bro_int_t` checks for `it >= 0` which should then always be true.

Should we instead first check for max `int64_t` (and possibly throw), and then create `bt` (possibly performing the replacement `-1` -> `std::numeric_limits<::bro_int_t>::max()`)?